packages/administration/src/Component/Datagrid/Adapter/Orm/ProxyQuery.php

```diff
@@ -0,0 +1,174 @@
+        $this->queryBuilder->resetDQLPart('select');
+    }
+
+    public function __clone()
+    {
+        $this->queryBuilder = clone $this->queryBuilder;
+    }
+
+    /**
+     * @param string $select
+     * @param string|null $alias
+     */
+    public function addSelect(string $select, ?string $alias = null): void
+    {
+        if ($alias !== null) {
+            $this->queryBuilder->addSelect($select . ' AS ' . $alias);
+
+            return;
+        }
+
+        $rootAlias = $this->rootAlias;
+
+        if (str_contains($select, '.') === false) {
+            $classMetadata = $this->entityManager->getClassMetadata($this->entityClass);
+
+            if (!in_array($select, $classMetadata->getFieldNames(), true)) {
+                throw new InvalidArgumentException('Field "' . $select . '" not found in entity ' . $this->entityClass);
+            }
+
+            $this->queryBuilder->addSelect($rootAlias . '.' . $select . ' AS ' . $select);
+
+            return;
+        }
+
+        $this->processDotNotation($select);
+    }
+
+    /**
+     * @param string $select
+     */
+    private function processDotNotation(string $select): void
+    {
+        $alias = $this->rootAlias;
+        $parts = explode('.', $select);
+
+        $currentClassMetadata = $this->entityManager->getClassMetadata($this->entityClass);
+
+        // dot notation is processed from left to right and each part is joined with left join
+        // the last part is added to select
+
+        foreach ($parts as $index => $part) {
+            $path = implode('.', array_slice($parts, 0, $index + 1));
+            $joinAlias = $part;
+
+
+            if ($index >= count($parts) - 1) {
+                if ($currentClassMetadata->hasField($part)) {
+                    $this->queryBuilder->addSelect("{$alias}.{$part}" . ' AS ' . $this->getAlias($path));
+
+                    continue;
+                }
+
+                try {
+                    // If entity does not have field, check if it has association to translations and try to select from translations
+                    $this->joinAssociation($currentClassMetadata, $path . '_tr', 'translations', $alias, $alias . '_tr');
+                    $this->queryBuilder->addSelect("{$alias}_tr.{$part}" . ' AS ' . $this->getAlias($path));
+
+                    continue;
+                } catch (InvalidArgumentException $exception) {
+                    // We don't need to throw exeption here
+                }
+
+                throw new InvalidArgumentException('Field "' . $part . '" not found in entity ' . $currentClassMetadata->getName());
+            }
+
+            $this->joinAssociation($currentClassMetadata, $path, $part, $alias, $joinAlias);
+
+            $currentClassMetadata = $this->entityManager->getClassMetadata(
+                $currentClassMetadata->getAssociationTargetClass($part),
+            );
+
+            $alias = $joinAlias;
```

Comment:
what is this good for in the last array iteration?

Reply:
completely refactored ProxyQuery :D